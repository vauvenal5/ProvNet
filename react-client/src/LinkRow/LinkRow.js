import React from 'react';
import { Label, Table, Button } from 'semantic-ui-react';
import TagView from '../TagView';

export const LinkRow = ({types, link}) => {
    
    let tags = link.tags.map(
        (tagId) => <TagView key={tagId} tag={types[tagId]}/>
    );

    return (
        <Table.Row>
            <Table.Cell>
                <Label as="a" ribbon>{link.address}</Label>
            </Table.Cell>
            <Table.Cell>{link.title}</Table.Cell>
            <Table.Cell>
                {tags}
            </Table.Cell>
            <Table.Cell>
                    <Button icon='info' content="Info" labelPosition='left'/>
                    <Button icon='eye' content="Select" labelPosition='left'/>
            </Table.Cell>
        </Table.Row>
    );
}