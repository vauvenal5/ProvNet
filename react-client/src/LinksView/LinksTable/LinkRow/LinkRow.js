import React from 'react';
import { connect } from 'react-redux';
import { Label, Table, Button } from 'semantic-ui-react';
import TagView from '../../../TagView';
import * as actions from '../../../modelActions';

export const LinkRow = ({types, link, linkSelect, contractSelect}) => {
    
    let tags = link.tags.map(
        (tagId) => <TagView key={tagId} tag={types[tagId]}/>
    );

    return (
        <Table.Row>
            <Table.Cell>
                {/* <Label as="a" ribbon 
                    onClick={(e, data) => linkSelect(link.getAddress())}
                > */}
                    {link.address}
                {/* </Label> */}
            </Table.Cell>
            <Table.Cell>{link.title}</Table.Cell>
            <Table.Cell>
                {tags}
            </Table.Cell>
            <Table.Cell>
                    {/* <Button icon='info' content="Info" labelPosition='left'/> */}
                    <Button onClick={() => linkSelect(link.getAddress())} icon='eye' content="Links" labelPosition='left'/>
                    <Button onClick={() => contractSelect(link.getAddress())} icon='file alternate outline' content="Select" labelPosition='left'/>
            </Table.Cell>
        </Table.Row>
    );
}

export const mapDispatchToProps = (dispatch) => {
    return {
        linkSelect: (address) => dispatch(actions.onLinkSelect(address)),
        contractSelect: (address) => dispatch(actions.onContractSelect(address))
    }
}

export default connect(undefined, mapDispatchToProps)(LinkRow);