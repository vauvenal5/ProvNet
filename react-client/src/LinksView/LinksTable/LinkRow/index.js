import Component from "./LinkRow";

export default Component;