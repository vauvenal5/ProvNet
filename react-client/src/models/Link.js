export default class Link {
    constructor(address="", tags=[], title="") {
        this.address = address;
        this.tags = tags;
        this.title = title;
    }

    getAddress() {
        return this.address;
    }

    getTitle() {
        return this.title;
    }

    getTags() {
        return this.tags;
    }
}