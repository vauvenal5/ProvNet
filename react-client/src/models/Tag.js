import _ from 'lodash';

const colors = [
    'red',
    'orange',
    'yellow',
    'olive',
    'green',
    'teal',
    'blue',
    'violet',
    'purple',
    'pink',
    'brown',
    'grey',
    'black',
];

export default class Tag {
    constructor(id, title) {
        this.id = id;
        this.title = title;
    }

    getColor() {
        return colors[(this.id-1)%colors.length];
    }

    getTitle() {
        return _.capitalize(this.title);
    }
}