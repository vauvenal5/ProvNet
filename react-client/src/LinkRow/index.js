import {LinkRow} from "./LinkRow";

export default LinkRow;