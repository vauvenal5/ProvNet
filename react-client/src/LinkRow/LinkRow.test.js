import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LinkRow from '.';
import Link from '../models/Link';
import Tag from '../models/Tag';

Enzyme.configure({ adapter: new Adapter() });

const setup = (propOverrides) => {
    const props = Object.assign({
        types: {
            ["1"]: new Tag(1, "trusted"),
            ["2"]: new Tag(2, "known")
        },
        link: new Link("linkAddress", [1,2], "tagTitle")
    }, propOverrides);

    const wrapper = shallow(<LinkRow {...props}/>);

    return {
        props,
        wrapper,
        render: wrapper.debug()
    }
};

describe("LinkRow", () => {
    it("should render", () => {
        const { render } = setup();
        expect(render).toMatchSnapshot();
    });
});