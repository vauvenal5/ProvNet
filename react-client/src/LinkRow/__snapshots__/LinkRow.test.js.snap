// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkRow should render 1`] = `
"<TableRow as=\\"tr\\" cellAs=\\"td\\">
  <TableCell as=\\"td\\">
    <Label as=\\"a\\" ribbon={true} onClick={[Function: onClick]}>
      linkAddress
    </Label>
  </TableCell>
  <TableCell as=\\"td\\">
    tagTitle
  </TableCell>
  <TableCell as=\\"td\\">
    <Component tag={{...}} />
    <Component tag={{...}} />
  </TableCell>
  <TableCell as=\\"td\\">
    <Button onClick={[Function: onClick]} icon=\\"eye\\" content=\\"Select\\" labelPosition=\\"left\\" as=\\"button\\" role=\\"button\\" />
  </TableCell>
</TableRow>"
`;
